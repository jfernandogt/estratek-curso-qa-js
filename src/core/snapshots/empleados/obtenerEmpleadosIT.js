const empleados = [
  {
    name: 'Fernando',
    department: 'IT',
  },
  {
    name: 'Ariel',
    department: 'Contabilidad',
  },
  {
    name: 'Carolina',
    department: 'IT',
  },
  {
    name: 'Luis',
    department: 'RRHH',
  },
  {
    name: 'Marcos',
    department: 'IT',
  },
]

export default function obtenerEmpleadosIT() {
  return empleados.filter((empleado) => empleado.department === 'IT')
}
