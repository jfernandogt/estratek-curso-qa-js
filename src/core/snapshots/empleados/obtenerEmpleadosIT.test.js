import obtenerEmpleadosIT from './obtenerEmpleadosIT'

test('Validar el listado de empleados de IT', () => {
  // Primero utilizamos console.log para obtenerlos
  const empleados = obtenerEmpleadosIT()
  // console.log(empleados)
  // expect(empleados).toEqual([
  //   { name: 'Fernando', department: 'IT' },
  //   { name: 'Carolina', department: 'IT' },
  // ])

  // La mejor forma es implementar un matchsnapshot
  expect(empleados).toMatchSnapshot()

  // Casos de uso
  /*
  1. Modelos de bases de datos
  2. Queries a bases de datos
  3. Templates
  4. Listados (zip codes, departamentos, paises)

  https://codewithhugo.com/abusing-jest-snapshot-tests-some-nice-use-cases/
  */
})

test('Validar el listado de empleados de IT usando inlineSnapshots', () => {
  const empleados = obtenerEmpleadosIT()
  expect(empleados).toMatchInlineSnapshot(`
    Array [
      Object {
        "department": "IT",
        "name": "Fernando",
      },
      Object {
        "department": "IT",
        "name": "Carolina",
      },
      Object {
        "department": "IT",
        "name": "Marcos",
      },
    ]
  `)
})
