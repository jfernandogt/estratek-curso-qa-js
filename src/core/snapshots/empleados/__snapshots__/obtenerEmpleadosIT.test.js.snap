// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validar el listado de empleados de IT 1`] = `
Array [
  Object {
    "department": "IT",
    "name": "Fernando",
  },
  Object {
    "department": "IT",
    "name": "Carolina",
  },
  Object {
    "department": "IT",
    "name": "Marcos",
  },
]
`;
